Add section for breaking changes to pull request template

andregasser opened the pull request:
# Description

I have added a section for listing breaking changes that have been introduced as part of a pull request.

# Type of Change

- Documentation only

# How Has This Been Tested?

No tests executed - documentation only.

# Mandatory Checklist

- [x] My change follows the projects coding style
- [ ] I ran `gradle check` and there were no errors reported
- [ ] I have performed a self-review of my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] KDoc added to all public methods

# Optional Things To Check

The items below are some more things to check before asking other people to review your code.

- In case you worked on new features: Did you also implement the reactive endpoint (bigbone-rx)?
- In case you added new *Methods classes: Did you also add it to `MastodonClient`?
- In case you worked on endpoints: Please mention in the PR that [API Coverage](https://github.com/andregasser/bigbone/wiki/Mastodon-API-Coverage) page needs to be updated (if needed)

PattaFeuFeu reviewed (approved):
Good change 👍 

Optional improvement: Should we remove the “Breaking change” item in “Type of change”? Otherwise, there’ll just be duplicate information in here.

andregasser commented:
> Good change 👍
> 
> Optional improvement: Should we remove the “Breaking change” item in “Type of change”? Otherwise, there’ll just be duplicate information in here.

Good idea! I have updated the template. Also added a better explanation on what we consider a breaking change (taken from the [Google OSS Library Breaking Change Policy](https://opensource.google/documentation/policies/library-breaking-change) guide).
